impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultDependencyResolverResult.java

```diff
@@ -102,19 +102,27 @@ public class DefaultDependencyResolverResult implements DependencyResolverResult
      * to {@link #addDependency(Node, Dependency, Predicate, Path)}.
      *
      * @param request the corresponding request
-     * @param cache cache of module information about each dependency
```

Comment:
`private final PathModularizationCache cache = new PathModularizationCache();` L:98

- https://www.reddit.com/r/learnjava/comments/ozgtxm/is_initializing_variables_at_their_declaration/
- https://softwareengineering.stackexchange.com/questions/379039/what-is-the-best-practice-for-initializing-variables-outside-or-inside-of-the-c
-

Reply:
In the particular case of this issue, creating the cache at the location where the field is defined is the opposite of the intend, because it makes the `PathModularizationCache` lifetime strongly coupled with the `DefaultDependencyResolverResult` lifetime. Such coupling makes `PathModularizationCache` close to useless. It was done that way in the previous version of this class only as a workaround for the fact that I didn't knew where to store a session-wide cache, but that workaround was intended to be temporary.

More details in a next comment.

Reply:
my suggest would not compile. im sorry.